app/models/piece.rb

```diff
@@ -40,4 +40,24 @@ def set_image
     color ? color_string = "white" : color_string = "black"
     self.image_name ||= "#{color_string}-#{piece_type.downcase}.png"
   end
+
+  def move_to!(x, y)
+    @target = game.pieces.where(:x_position => x, :y_position => y).take
+    if @target.nil?
```

Comment:
I thought about that and rubocop won't be happy; the other solution is to repeat `valid_move` twice.